### ISSUE TITLE
Font weight doesn't work

### ISSUE THREAD

**Issue by seilem** [Automattic/node-canvas#1452]
## Issue
I tried to use different font weight and it doesn't work right.

## Steps to Reproduce

```js
var fs = require('fs')
var path = require('path')

const { createCanvas, registerFont } = require('canvas')

function fontFile (name) {
  return path.join(__dirname, '/files/fonts/', name)
}

// ExtraBold first.
registerFont(fontFile('OpenSans-ExtraBold.ttf'), { family: 'OpenSans', weight: '800' })
registerFont(fontFile('OpenSans-SemiBold.ttf'), { family: 'OpenSans', weight: '600' })
registerFont(fontFile('OpenSans-Bold.ttf'), { family: 'OpenSans', weight: 'bold' })

var canvas = createCanvas(500, 500)
var ctx = canvas.getContext('2d')

ctx.font = '600 50px OpenSans'
ctx.fillText('Hello?', 0, 60)

ctx.font = 'bold 50px OpenSans'
ctx.fillText('Hello?', 0, 60 * 2)

ctx.font = '800 50px OpenSans'
ctx.fillText('Hello?', 0, 60 * 3)

canvas.createPNGStream().pipe(fs.createWriteStream(path.join(__dirname, 'font.png')))
```
![font](https://user-images.githubusercontent.com/3086319/60543040-03cf2e80-9d1e-11e9-8854-7e3d67a58882.png)

Weight depends on a first registered font.

```js
var fs = require('fs')
var path = require('path')

const { createCanvas, registerFont } = require('canvas')

function fontFile (name) {
  return path.join(__dirname, '/files/fonts/', name)
}

// SemiBold first.
registerFont(fontFile('OpenSans-SemiBold.ttf'), { family: 'OpenSans', weight: '600' })
registerFont(fontFile('OpenSans-Bold.ttf'), { family: 'OpenSans', weight: 'bold' })
registerFont(fontFile('OpenSans-ExtraBold.ttf'), { family: 'OpenSans', weight: '800' })


var canvas = createCanvas(250, 250)
var ctx = canvas.getContext('2d')

ctx.font = '600 50px OpenSans'
ctx.fillText('Hello?', 0, 60)

ctx.font = 'bold 50px OpenSans'
ctx.fillText('Hello?', 0, 60 * 2)

ctx.font = '800 50px OpenSans'
ctx.fillText('Hello?', 0, 60 * 3)

canvas.createPNGStream().pipe(fs.createWriteStream(path.join(__dirname, 'font.png')))
```
![font](https://user-images.githubusercontent.com/3086319/60543229-70e2c400-9d1e-11e9-876f-c6cbe7105b19.png)

## Your Environment
* Version of node-canvas: canvas@2.5.0 
* Environments:macOS Mojave 10.14.5 and Debian GNU/Linux 8 (jessie)

**Comment by luankefei** [comment]
+1

**Comment by jrvansuita** [comment]
+1

**Comment by asturur** [comment]
Just wanted to report i m on mojave 10.14.6 and for me the example work good on node-canvas 2.6.1

![image](https://user-images.githubusercontent.com/1194048/67640052-cc514700-f8f7-11e9-98e5-bc3b0b01a384.png)

**Comment by LinusU** [comment]
That's amazing!

If anyone still is having troubles on the latest version, I'd be happy to reopen ☺️

**Comment by Sprocket06** [comment]
On windows 10, using the latest version (node-canvas 2.6.1), the above example still fails to differentiate weights.
![font](https://user-images.githubusercontent.com/24927632/98592156-4c4a1500-229f-11eb-87b0-728737b3781d.png)

**Comment by manavpeerbits** [comment]
Hello @LinusU 

I'm having the same issue
Bold text style is not applied in my case also

` bold 56px "Helvetica Neue"`

While Italic is working
`italic 37px "Helvetica Neue"`

My Environment

MAC OS Catalina 10.15.7
Canvas 2.8.0

**Comment by AmreshSinha** [comment]
> Hello @LinusU
> 
> I'm having the same issue
> Bold text style is not applied in my case also
> 
> ` bold 56px "Helvetica Neue"`
> 
> While Italic is working
> `italic 37px "Helvetica Neue"`
> 
> My Environment
> 
> MAC OS Catalina 10.15.7
> Canvas 2.8.0

It's working fine my side.
My environment:
windows 10, node version 14.16.1
canvas v2.8.0

**Comment by colefranz** [comment]
To anyone coming in with the same issue, I was able to workaround this issue by registering the fonts under different `family`s. E.g. I used `OpenSans`, `OpenSansBold`, and `OpenSansItalic`. It's rather unfortunate but it does seem to get the job done.

**Comment by konser80** [comment]
We still need a fix. 
bold doesn't work

node-canvas 2.11.2, node 18.18.2, os x 14.0

**Comment by pranavburnwal** [comment]
Any possibilities of help on this one?

**Comment by konser80** [comment]
workaround - is to register bold fonts by different names, for ex:

```
name: "HelveticaBold"
weight: "regular"
```

**Comment by kostia1st** [comment]
For me, font weight does **not** work as well, in both Linux (bookworm) and Windows 11. 
The font is _variable_ (Roboto Flex) and I got it installed into the OS, and additionally attempted to get installed thru `registerFont`. Tried multiple different approaches, including the one claimed above as "workaround" - nothing works. 
When I try to use that same font in say WordPad and make it bold - WordPad displays bold as expected. So it's clearly issue with `node-canvas` or its dependencies, not the font itself.
